pkg/kinds/dashboard/compose.go

```diff
@@ -0,0 +1,163 @@
+package dashboard
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"cuelang.org/go/cue"
+	cjson "cuelang.org/go/encoding/json"
+	"github.com/grafana/kindsys"
+)
+
+func (k *Kind) Compose(slot kindsys.Slot, kinds ...kindsys.Composable) (kindsys.Core, error) {
```

Comment:
Shouldn't be use directly [kindsys](https://github.com/grafana/kindsys/pull/31/files#diff-78df076792d2653188e9160a19179005e0571ca7262db8f6464965220090c778R68)? Or is that temporal until we merge it?

Reply:
yeah i went back and forth on whether we should duplicate the logic here that also exists over there. What's here is supposed to be temporary.

The only reason i duplicated it over here is because i hadn't made a method that allowed getting access to that `composed` map, so it wasn't possible to do this hacky `Validate()` over here.

We could probably just add a method in kindsys, maybe `Composed() map[Slot]Composable` or something, that copies and returns that `composed` map. Then we can get rid of the duplication on this side.